From 2778d5fe5116f206af760c57239537b7cfce0caa Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Fri, 27 Jul 2012 06:27:47 -0400
Subject: [PATCH] Add legacy pam-foreground-compat.ck to enable
 /var/run/console

Taken from:
https://projects.archlinux.org/svntogit/packages.git/commit/trunk/PKGBUILD?h=packages/consolekit&id=e72abef5f88090a600a32216661fbd0a35a65414

We need /var/run/console to have legacy DBus policies work, etc.  This
file can go away when switching to systemd.
---
 Makefile-legacy-console.am |    3 +++
 Makefile.am                |    2 ++
 pam-foreground-compat.ck   |   16 ++++++++++++++++
 3 files changed, 21 insertions(+)
 create mode 100644 Makefile-legacy-console.am
 create mode 100644 pam-foreground-compat.ck

diff --git a/Makefile-legacy-console.am b/Makefile-legacy-console.am
new file mode 100644
index 0000000..f167b35
--- /dev/null
+++ b/Makefile-legacy-console.am
@@ -0,0 +1,3 @@
+
+pamforegroundcompat_DATA = pam-foreground-compat.ck
+pamforegroundcompatdir = $(libdir)/ConsoleKit/run-session.d
diff --git a/Makefile.am b/Makefile.am
index 9f13e2c..ba12e0e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -66,3 +66,5 @@ $(srcdir)/ChangeLog:
 
 DISTCHECK_CONFIGURE_FLAGS = \
 	--with-systemdsystemunitdir=$$dc_install_base/$(systemdsystemunitdir)
+
+include Makefile-legacy-console.am
diff --git a/pam-foreground-compat.ck b/pam-foreground-compat.ck
new file mode 100644
index 0000000..7f6b122
--- /dev/null
+++ b/pam-foreground-compat.ck
@@ -0,0 +1,16 @@
+#!/bin/sh
+TAGDIR=/var/run/console
+
+[ -n "$CK_SESSION_USER_UID" ] || exit 1
+
+TAGFILE="$TAGDIR/`getent passwd $CK_SESSION_USER_UID | cut -f 1 -d:`"
+
+if [ "$1" = "session_added" ]; then
+    mkdir -p "$TAGDIR"
+    echo "$CK_SESSION_ID" >> "$TAGFILE"
+fi
+
+if [ "$1" = "session_removed" ] && [ -e "$TAGFILE" ]; then
+    sed -i "\%^$CK_SESSION_ID\$%d" "$TAGFILE"
+    [ -s "$TAGFILE" ] || rm -f "$TAGFILE"
+fi
-- 
1.7.10.4

